software/webapp/src/main/java/org/apache/brooklyn/entity/webapp/JavaWebAppSshDriver.java

```diff
@@ -93,11 +93,11 @@ protected String inferRootUrl() {
         if (isProtocolEnabled("https")) {
             Integer port = getHttpsPort();
             checkNotNull(port, "HTTPS_PORT sensors not set; is an acceptable port available?");
-            return String.format("https://%s:%s/", getSubnetHostname(), port);
+            return String.format("https://%s:%s/", getHostname(), port);
```

Comment:
@grkvlt will this cause a problem from Clocker?

The problem is it depends on the context for whether you want the subnet hostname/ip or the public hostname/ip. We don't know at this point how the main.uri sensor is going to be used.